lib/game/entity.go

```diff
@@ -5,37 +5,74 @@ import (
 	"buildblast/lib/physics"
 )
 
-//Interface currently implemented by Player (not to say anything cannot
-//implement it, but for now it is just implemented by player).
+// Should be an int someday...
+type EntityId string
+
+type EntityKind string
+
+const (
+	EntityKindPlayer    = EntityKind("player")
+	EntityKindWorldItem = EntityKind("worldItem")
+)
+
 type Entity interface {
-	Pos() coords.World
-	Look() coords.Direction
-	Health() int
-	Vy() float64
+	EntityId() EntityId
+	Body() physics.Body
 	LastUpdated() float64
-	State() EntityState
-
-	Tick(w *World)
-	Damage(amount int)
-	Dead() bool
-	Respawn(pos coords.World)
+	Wpos() coords.World
+	Look() coords.Direction
 	BoxAt(t float64) *physics.Box
-	ID() EntityID
 }
 
 type EntityState struct {
-	Pos       coords.World
-	Look      coords.Direction
-	Health    int
-	Vy        float64
-	Timestamp float64
+	EntityId    EntityId
+	Body        physics.Body
+	LastUpdated float64
 }
 
-// Should be an int someday...
-type EntityID string
+func (es *EntityState) Wpos() coords.World {
+	return coords.World(es.Body.Pos)
+}
 
-type EntityKind string
+func (es *EntityState) Look() coords.Direction {
+	return coords.Direction(es.Body.Dir)
+}
 
-const (
-	EntityKindPlayer = EntityKind("player")
-)
+type Damageable interface {
+	Life() int
+	Dead() bool
+
+	Damage(amount int)
+}
+
+type Health struct {
+	Life int
+}
+
+type Possessor interface {
+	Inventory() *Inventory
+	Give(item Item) bool
+	Take(item Item) bool
+
+	// Do items in the world gravitate towards this?
+	Collects() bool
+	// If Collects return true, then Body must return a value
+	Body() physics.Body
```

Comment:
I'm going to go with... no? I can't think of anything off the top of my head. Anything I can think of would have a body. Unless we had a black hole type thing. Like garbage disposal. But even that can be abstracted to have a body.